diff -Naur /etc/apparmor.d.orig/abstractions/base /etc/apparmor.d/abstractions/base
--- a/etc/apparmor.d/abstractions/base	2025-04-02 16:15:24.000000000 +0000
+++ b/etc/apparmor.d/abstractions/base	2025-04-15 15:02:55.712037894 +0000
@@ -33,6 +33,7 @@
   @{etc_ro}/locale.alias       r,
   @{etc_ro}/localtime          r,
   @{etc_rw}/localtime          r,
+  /etc/tor/torsocks.conf         r,
   /etc/writable/localtime        r,
   /usr/share/locale-bundle/**    r,
   /usr/share/locale-langpack/**  r,
@@ -73,9 +74,9 @@
   /opt/*-linux-uclibc/lib/ld-uClibc*so* mr,
 
   # we might as well allow everything to use common libraries
-  /{usr/,}lib{,32,64}/**                r,
-  /{usr/,}lib{,32,64}/**.so*       mr,
-  /{usr/,}lib/@{multiarch}/**            r,
+  /{usr/,}lib{,32,64}/{[^l],l[^i],li[^v],liv[^e],live[^/]}**           r,
+  /{usr/,}lib{,32,64}/{[^l],l[^i],li[^v],liv[^e],live[^/]}**.so*       mr,
+  /{usr/,}lib/@{multiarch}/{[^l],l[^i],li[^v],liv[^e],live[^/]}**      r,
   /{usr/,}lib/@{multiarch}/**.so*   mr,
   /{usr/,}lib/tls/i686/{cmov,nosegneg}/*.so*    mr,
   /{usr/,}lib/i386-linux-gnu/tls/i686/{cmov,nosegneg}/*.so*    mr,
diff -Naur /etc/apparmor.d/abstractions.orig/ubuntu-helpers /etc/apparmor.d/abstractions/ubuntu-helpers
--- /etc/apparmor.d/abstractions.orig/ubuntu-helpers	2025-04-10 15:06:25.000000000 +0000
+++ /etc/apparmor.d/abstractions/ubuntu-helpers	2025-05-01 12:00:17.520861277 +0000
@@ -52,10 +52,11 @@
 
   # Allow exec of anything, but under this profile. Allow transition
   # to other profiles if they exist.
-  /{usr/,usr/local/,}{bin,sbin}/* Pixr,
+  /{usr/,usr/local/,}{bin,sbin}/{[^hu],h[^p],hp[^i],hpi[^j],hpij[^s],u[^n],un[^i],uni[^x],unix[^_],unix_[^c],unix_c[^h],unix_ch[^k],unix_chk[^p],unix_chkp[^w],unix_chkpw[^d]}* Pixr,
 
   # Allow exec of libexec applications in /usr/lib* and /usr/local/lib*
-  /usr/{,local/}lib*/{,**/}* Pixr,
+  /usr/{,local/}lib*/* Pixr,
+  /usr/{,local/}lib*/{[^l],l[^i],li[^v],liv[^e],live[^/]}{,**/}* Pixr,
 
   # Allow exec of software-center scripts. We may need to allow wider
   # permissions for /usr/share, but for now just do this. (LP: #972367)
@@ -68,8 +69,8 @@
   # in limited libraries so glibc's secure execution should be enough to not
   # require the santized_helper (ie, LD_PRELOAD will only use standard system
   # paths (man ld.so)).
-  /usr/lib/chromium-browser/chromium-browser-sandbox PUxr,
-  /usr/lib/chromium{,-browser}/chrome-sandbox PUxr,
+  # /usr/lib/chromium-browser/chromium-browser-sandbox PUxr,
+  # /usr/lib/chromium{,-browser}/chrome-sandbox PUxr,
   /opt/google/chrome{,-beta,-unstable}/chrome-sandbox PUxr,
   /opt/google/chrome{,-beta,-unstable}/google-chrome Pixr,
   /opt/google/chrome{,-beta,-unstable}/chrome Pixr,
@@ -85,12 +86,13 @@
 
   # Allow running snap applications
   # https://bugs.launchpad.net/apparmor/+bug/2095872
-  /usr/bin/snap ux,
+  # /usr/bin/snap ux,
 
   # Full access
   / r,
   /** rwkl,
-  /{,usr/,usr/local/}lib{,32,64}/{,**/}*.so{,.*} m,
+  /lib/{[^l],l[^i],li[^v],liv[^e],live[^/]}{,**/}*.so{,.*} m,
+  /usr{/,/local/}lib{,32,64}/{,**/}*.so{,.*} m,
 
   # Dangerous files
   audit deny owner /**/* m,              # compiled libraries
