--- /usr/lib/python3/dist-packages/gsecrets/widgets/window.py	2025-04-05 08:15:13.000000000 +0200
+++ /usr/lib/python3/dist-packages/gsecrets/widgets/window.py	2026-01-29 12:47:26.440331482 +0100
@@ -238,8 +238,10 @@
         dialog.props.title = _("Create Safe")
         dialog.props.filters = filters
         dialog.props.accept_label = _("_Create")
-        # NOTE: Safe as in strongbox
-        dialog.props.initial_name = _("Safe") + ".kdbx"
+        dialog.props.initial_name = "Passwords.kdbx"
+        default_dir = Gio.File.new_for_path("/home/amnesia/Persistent/")
+        if default_dir.query_exists():
+            dialog.set_initial_folder(default_dir)
 
         dialog.save(self, None, self._on_create_filechooser_response)
 
