diff --git a/src/jsc/bindings/BunProcess.cpp b/src/jsc/bindings/BunProcess.cpp
index 6935f0c..56da7f0 100644
--- a/src/jsc/bindings/BunProcess.cpp
+++ b/src/jsc/bindings/BunProcess.cpp
@@ -183,6 +183,8 @@ static JSValue constructArch(VM& vm, JSObject* processObject)
     return JSC::jsString(vm, makeAtomString("x64"_s));
 #elif CPU(ARM64)
     return JSC::jsString(vm, makeAtomString("arm64"_s));
+#elif CPU(RISCV64)
+    return JSC::jsString(vm, makeAtomString("riscv64"_s));
 #else
 #error "Unknown architecture"
 #endif
@@ -2556,6 +2558,9 @@ static JSValue constructProcessConfigObject(VM& vm, JSObject* processObject)
 #elif CPU(ARM64)
     variables->putDirect(vm, JSC::Identifier::fromString(vm, "host_arch"_s), JSC::jsString(vm, String("arm64"_s)), 0);
     variables->putDirect(vm, JSC::Identifier::fromString(vm, "target_arch"_s), JSC::jsString(vm, String("arm64"_s)), 0);
+#elif CPU(RISCV64)
+    variables->putDirect(vm, JSC::Identifier::fromString(vm, "host_arch"_s), JSC::jsString(vm, String("riscv64"_s)), 0);
+    variables->putDirect(vm, JSC::Identifier::fromString(vm, "target_arch"_s), JSC::jsString(vm, String("riscv64"_s)), 0);
 #else
 #error "Unsupported architecture"
 #endif
