diff --git a/src/jsc/bindings/ConsoleObject.cpp b/src/jsc/bindings/ConsoleObject.cpp
index 6d8fdda..95b6e5f 100644
--- a/src/jsc/bindings/ConsoleObject.cpp
+++ b/src/jsc/bindings/ConsoleObject.cpp
@@ -34,11 +34,13 @@ void ConsoleObject::messageWithTypeAndLevel(MessageType type, MessageLevel level
     JSC::JSGlobalObject* globalObject,
     Ref<ScriptArguments>&& arguments)
 {
+#if !(defined(__riscv) && __riscv_xlen == 64)
     if (globalObject->inspectable()) {
         if (auto client = globalObject->inspectorController().consoleClient()) {
             client->messageWithTypeAndLevel(type, level, globalObject, arguments.copyRef());
         }
     }
+#endif
     auto& vm = JSC::getVM(globalObject);
     auto args = arguments.ptr();
     JSC::EncodedJSValue jsArgs[255];
@@ -112,20 +114,24 @@ void ConsoleObject::warnUnimplemented(const String& method) {}
 
 void ConsoleObject::profile(JSC::JSGlobalObject* globalObject, const String& title)
 {
+#if !(defined(__riscv) && __riscv_xlen == 64)
     if (globalObject->inspectable()) {
         if (auto client = globalObject->inspectorController().consoleClient()) {
             client->profile(globalObject, title);
         }
     }
+#endif
 }
 
 void ConsoleObject::profileEnd(JSC::JSGlobalObject* globalObject, const String& title)
 {
+#if !(defined(__riscv) && __riscv_xlen == 64)
     if (globalObject->inspectable()) {
         if (auto client = globalObject->inspectorController().consoleClient()) {
             client->profileEnd(globalObject, title);
         }
     }
+#endif
 }
 
 }
